packages/mobile/src/transactions/saga.ts

```diff
@@ -1,25 +1,130 @@
+import { call, put, select, spawn, take } from 'redux-saga/effects'
 import { showError } from 'src/alert/actions'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
+import { apolloClient } from 'src/apollo'
+import { Token, TransactionFeedFragment, UserTransactionsQuery } from 'src/apollo/types'
 import { ErrorMessages } from 'src/app/ErrorMessages'
 import { CURRENCY_ENUM } from 'src/geth/consts'
 import { fetchGoldBalance } from 'src/goldToken/actions'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
 import { fetchDollarBalance } from 'src/stableToken/actions'
 import {
   Actions,
   addHashToStandbyTransaction,
+  newTransactionsInFeed,
   removeStandbyTransaction,
   transactionConfirmed,
   transactionFailed,
 } from 'src/transactions/actions'
 import { TxPromises } from 'src/transactions/contract-utils'
+import {
+  knownFeedTransactionsSelector,
+  KnownFeedTransactionsType,
+  standbyTransactionsSelector,
+} from 'src/transactions/reducer'
 import { sendTransactionPromises, wrapSendTransactionWithRetry } from 'src/transactions/send'
+import { getTxsFromUserTxQuery } from 'src/transactions/transferFeedUtils'
+import { StandbyTransaction, TransactionStatus } from 'src/transactions/types'
 import Logger from 'src/utils/Logger'
+import { getAccount } from 'src/web3/saga'
 
 const TAG = 'transactions/saga'
 
+interface UserTxQueryChannelEvent {
+  transactionFeedFragments: TransactionFeedFragment[]
+}
+
+export function* initializeUserTxListQueryWatcher() {
+  const address: string | null = yield call(getAccount)
+  const localCurrencyCode = yield select(getLocalCurrencyCode)
+  const token = Token.CUsd
+
+  // Using a dynamic require import here because of an import cycle around TransactionsList that is
+  // causing many tests to fail. Alternative is to mock this entire saga file or significantly restructure things.
+  const query = require('src/transactions/TransactionsList').TRANSACTIONS_QUERY
+
+  const userTxQueryChannel: EventChannel<UserTxQueryChannelEvent> = eventChannel((emitter) => {
+    const querySubscription = apolloClient
+      .watchQuery<UserTransactionsQuery | undefined>({
+        query,
+        variables: {
+          address,
+          token,
+          localCurrencyCode,
+        },
+        fetchPolicy: 'cache-and-network',
+      })
+      .subscribe({
+        next: (queryResult) => {
+          if (queryResult.loading) {
+            return
+          }
+
+          if (queryResult.errors) {
+            Logger.error(TAG, 'Error watching user tx query' + JSON.stringify(queryResult.errors))
+            return
+          }
+          emitter({
+            transactionFeedFragments: getTxsFromUserTxQuery(queryResult.data),
+          })
+        },
+        error: (e) => console.error(e),
+      })
+
+    return querySubscription.unsubscribe
+  })
+  yield spawn(watchUserTxQueryChannel, userTxQueryChannel)
+}
```

Comment:
This feels disconnected from what's happening at the component level.
It looks like it's working for both cUSD and cGLD feeds because there's a link between them.
But this is gonna create differences, and probably one extra query when the local currency is changed for instance.

What do you think of instead of trying to replicate the query here we would dispatch the new transactions from the `onComplete` function at the component level in `TransactionsList`?

Reply:
Can you clarify what you mean by creating differences? I don't follow.

The problem with the onComplete is that
1) It seems to be less reliable than this watcher
2) It forces us to pollute the component with this business logic

As for creating a new query, this will pull data from the cache first so if the component already queried, it won't use additional bandwidth.

Reply:
To clarify what I mean by unreliable, issues #2647, #1335,  #3624 and #2704 shouldn't be happening if the onComplete was working as expected. My suspicion in that only having it run when the component is mounted contributes to these issues.